circuits/web/parsers/http.py

```diff
@@ -187,7 +177,9 @@ def execute(self, data, length):
                 else:
                     self.__on_firstline = True
                     self._buf.append(data[:idx])
-                    first_line = bytes_to_str(b("").join(self._buf))
+                    first_line = b("").join(self._buf)
+                    if PY3:
+                        first_line = str(first_line, 'unicode_escape')
```

Comment:
Okay, this takes a little bit -- lets step back:

What did `bytes_to_str(b("").join(self._buf))` originally do:
On py2 `bytes_to_str` is a no-op: https://github.com/circuits/circuits/blob/5797ea3769d8197a0ca3a4c7db898a740ce11a89/circuits/six.py#L299-L300
hence leaving it out is okay for python 2.

On Py3 it did decode the bytes using `unicode_escape`:
https://github.com/circuits/circuits/blob/5797ea3769d8197a0ca3a4c7db898a740ce11a89/circuits/six.py#L281-L282
which my `if PY3` should do.

(`bytes_to_str` is no longer part [or rather was never] of six, hence I removed it when updating six to upstream)

Arguably the code is broken (I am not sure that unicode_escape is the way to go here), but it is the same brokenness as before (or same correctness for that matter)